db/migrate/20210407113051_add_username_to_users.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddUsernameToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :username, :string
```

Comment:
You'll need to index this

Reply:
Why would we need an index here? Do we expect to get users by their username?

Reply:
Do we have a requirement regarding username uniqueness? It would be unusual if the usernames were not unique. But I take your point, we are not currently looking up the current user by username.

Reply:
I think this is meant to be a 'preferred name', so I think we won't worry about uniqueness.

Many people would like to be referred to as 'Joe'.

Maybe we should call it `preferred_name` in the database, and `displayed_name` in the model method.

Reply:
I don't think it should be called username _anywhere_ (it's causing confusion already and we're the ones developing it), but I think if it's called username for the client it should be called username internally too.

If it's too confusing for us, we shouldn't be using that term for users either.